code/datums/elements/food/processable.dm

```diff
@@ -0,0 +1,33 @@
+// If an item has the processable item, it can be processed into another item with a specific tool. This adds generic behavior for those actions to make it easier to set-up generically.
+/datum/element/processable
+	element_flags = ELEMENT_BESPOKE
+	id_arg_index = 2
+	///The type of atom this creates when the processing recipe is used.
+	var/result_atom_type
+	///The tool behaviour for this processing recipe
+	var/tool_behaviour
+	///Time to process the atom
+	var/time_to_process
+	///Amount of the resulting actor this will create
+	var/amount_created
+
+/datum/element/processable/Attach(datum/target, tool_behaviour, result_atom_type, amount_created = 3, time_to_process = 20)
+	SIGNAL_HANDLER
```

Comment:
`SIGNAL_HANDLER` needs to be defined at the base of the proc.
What signal is calling this directly? This is not kosher.

Reply:
accidente